azkaban-common/src/test/java/azkaban/trigger/BasicTimeCheckerTest.java

```diff
@@ -77,6 +78,108 @@ public void basicTimerTest() {
     }
 
     assertTrue(cond.isMet());
+  }
+
+  /**
+   * Test Base Cron Functionality.
+   */
+  @Test
+  public void cronTimerTest1() {
+
+    Map<String, ConditionChecker> checkers =
+        new HashMap<String, ConditionChecker>();
+
+    DateTime now = DateTime.now();
+    String cronExpression = "0 0 0 31 12 ? 2050";
+
+    BasicTimeChecker timeChecker =
+        new BasicTimeChecker("BasicTimeChecket_1", now.getMillis(),
+            now.getZone(), true, true, null, cronExpression);
+    System.out.println("getNextCheckTime = " + timeChecker.getNextCheckTime());
+
+    checkers.put(timeChecker.getId(), timeChecker);
+    String expr = timeChecker.getId() + ".eval()";
+    Condition cond = new Condition(checkers, expr);
+    // 2556086400000L represent for "2050-12-31T00:00:00.000-08:00"
+
+    DateTime year2050 = new DateTime(2050, 12, 31, 0 ,0 ,0 ,now.getZone());
+    assertTrue(cond.getNextCheckTime() == year2050.getMillis());
+  }
+
+  /**
+   * Test when PST-->PDT happens in 2020. -8:00 -> -7:00
+   * See details why confusion happens during this change: https://en.wikipedia.org/wiki/Pacific_Time_Zone
```

Comment:
Add more details? The next execution time will become 3:30 PDT instead of 2:30 PST before.